### PR TITLE
Changed code to improve output for files under test/e2e/apimachinery

### LINKED ISSUE #105678: replace ExpectEqual(someBoolean, ...)
### What would you like to be added?

The `test/e2e` tests are full of assertions which compare a boolean against true or false, often without any additional explanation. When those assertions fail, the error message is useless for understanding what went wrong, often just saying:
```
      Apr 22 09:03:41.130: Expected
          <bool>: false
      to equal
          <bool>: true
```

Including an explanation is better, but then we still have the distracting noise about the comparison:
```
      Apr 22 09:08:49.598: Failed to do XYZ
      Expected
          <bool>: false
      to equal
          <bool>: true
```

All of those assertions should be replace with `if (<failure check>) framework.Failf(<informative message>)`.

Such locations can be found by grepping:
```
$ git grep 'Expect.*Equal([^,]*, *\(true\|false\)' | wc -l
272
```

Some progress has been made in the meantime:
- v1.23.0: 250
- v1.24.0-beta.0: 177

However, we were down to 175 at some point, so the number is going back up.

Example:
```
test/e2e/storage/vsphere/vsphere_volume_diskformat.go:  framework.ExpectEqual(isAttached, true)
=>
if (!isAttached) {
   framework.Failf("volume %s is not attached", ...)
}
```

Some other places compare values. Those should be replaced with a comparison assertion. Example:
```
test/e2e/upgrades/apps/etcd.go: framework.ExpectEqual(ratio > 0.75, true)
=>
gomega.Expect(ratio).To(gomega.BeNumerically(">", 0.75), "ratio too small")
```

It makes sense to split this work into individual PRs, one per OWNERS file under test/e2e. Here's a list of directories:
- [ ] test/e2e/apimachinery @hunknownz
- [x] test/e2e/apps: https://github.com/kubernetes/kubernetes/pull/109944
- [x] test/e2e/auth
- [x] test/e2e/autoscaling: https://github.com/kubernetes/kubernetes/pull/106200
- [x] test/e2e/cloud/gcp
- [x] test/e2e/cloud
- [x] test/e2e/common/network
- [x] test/e2e/common/node: @jwtty
- [x] test/e2e/common/storage: https://github.com/kubernetes/kubernetes/pull/106624
- [x] test/e2e/framework: https://github.com/kubernetes/kubernetes/pull/105939
- [x] test/e2e/framework/providers/vsphere
- [x] test/e2e/framework/volume
- [x] test/e2e/instrumentation/logging
- [x] test/e2e/instrumentation: https://github.com/kubernetes/kubernetes/pull/106233
- [x] test/e2e/kubectl
- [x] test/e2e/lifecycle
- [x] test/e2e/node: https://github.com/kubernetes/kubernetes/pull/106486
- [x] test/e2e/scheduling: https://github.com/kubernetes/kubernetes/pull/106486
- [x] test/e2e/storage: @SD-13 (prior PR: https://github.com/kubernetes/kubernetes/pull/105860)
- [x] test/e2e/testing-manifests/storage-csi
- [x] test/e2e/upgrades/apps: https://github.com/kubernetes/kubernetes/pull/105774
- [x] test/e2e/windows: https://github.com/kubernetes/kubernetes/pull/106220, https://github.com/kubernetes/kubernetes/pull/110303

When creating a PR, use `<directory>: enhance assertions` as title. Use one commit per PR, squash when addressing feedback. Unless noted otherwise, @chetak123 will work on PRs. Help may be welcome.

/sig testing
/help

### Why is this needed?

Better output for developers when tests fail.

### PR DESCRIPTION
**Pull request by NikhilSharmaWe**
#### What type of PR is this?
Enhancement 

#### What this PR does / why we need it:
For better (more informative) output for developers when test fails. Changed files are under test/e2e/apimachinery for this PR.
#### Which issue(s) this PR fixes:

Fixes #105678

#### Does this PR introduce a user-facing change?

```release-note
none
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/e2e/apimachinery/custom_resource_definition.go [hunk 1]
```diff
@@ -339,7 +339,9 @@ var _ = SIGDescribe("CustomResourceDefinition resources [Privileged:ClusterAdmin
 		}}, metav1.CreateOptions{})
 		framework.ExpectNoError(err, "creating CR")
 		v, found, err := unstructured.NestedFieldNoCopy(u2.Object, "a")
-		framework.ExpectEqual(found, true, "\"a\" is defaulted")
+		if !found{
+			framework.Failf("value of object in %+v with field a is nil or it does not exists or its type is not expected type map[string]interface", u2.Object)
```
Comment (NikhilSharmaWe) [inline test/e2e/apimachinery/custom_resource_definition.go:343]: @pohly not very sure about this output, could you advise a better format of output here.

Comment (pohly) [reply test/e2e/apimachinery/custom_resource_definition.go:343]:
```suggestion
			framework.Failf("field `a` should have been defaulted in %+v", u2.Object)
```

A developer reading this should know what "defaulted" means. It also was what the original message said.

File: test/e2e/apimachinery/aggregator.go [hunk 2]
```diff
@@ -512,7 +512,9 @@ func TestSampleAPIServer(f *framework.Framework, aggrclient *aggregatorclient.Cl
 			break
 		}
 	}
-	framework.ExpectEqual(locatedWardle, true, "Unable to find v1alpha1.wardle.example.com in APIServiceList")
+	if !locatedWardle{
```
Comment (pohly) [inline test/e2e/apimachinery/aggregator.go:515]: Check with `gofmt`, this looks wrong.

File: test/e2e/apimachinery/resource_quota.go [hunk 3]
```diff
@@ -909,7 +909,9 @@ var _ = SIGDescribe("ResourceQuota", func() {
 
 		ginkgo.By("Verifying the deleted ResourceQuota")
 		_, err = client.CoreV1().ResourceQuotas(ns).Get(context.TODO(), quotaName, metav1.GetOptions{})
-		framework.ExpectEqual(apierrors.IsNotFound(err), true)
+		if !apierrors.IsNotFound(err){
+			framework.Failf("Error:", err)
```
Comment (pohly) [inline test/e2e/apimachinery/resource_quota.go:913]:
```suggestion
			framework.Failf("Expected `not found` error, got: %v", err)
```

File: test/e2e/apimachinery/resource_quota.go [hunk 4]
```diff
@@ -1099,7 +1101,9 @@ var _ = SIGDescribe("ResourceQuota [Feature:PodPriority]", func() {
 	ginkgo.It("should verify ResourceQuota's priority class scope (quota set to pod count: 1) against a pod with same priority class.", func() {
 
 		_, err := f.ClientSet.SchedulingV1().PriorityClasses().Create(context.TODO(), &schedulingv1.PriorityClass{ObjectMeta: metav1.ObjectMeta{Name: "pclass1"}, Value: int32(1000)}, metav1.CreateOptions{})
-		framework.ExpectEqual(err == nil || apierrors.IsAlreadyExists(err), true)
+		if err == nil && apierrors.IsAlreadyExists(err){
```
Comment (pohly) [inline test/e2e/apimachinery/resource_quota.go:1104]: This isn't the same logic. The following expressions are equivalent:

```
err == nil || apierrors.IsAlreadyExists(err) == true
!(err == nil || apierrors.IsAlreadyExists(err))
err != nil && !apierrors.IsAlreadyExists(err)
```

The last one is what we want (simplest).

File: test/e2e/apimachinery/resource_quota.go [hunk 5]
```diff
@@ -1138,7 +1142,9 @@ var _ = SIGDescribe("ResourceQuota [Feature:PodPriority]", func() {
 	ginkgo.It("should verify ResourceQuota's priority class scope (quota set to pod count: 1) against 2 pods with same priority class.", func() {
 
 		_, err := f.ClientSet.SchedulingV1().PriorityClasses().Create(context.TODO(), &schedulingv1.PriorityClass{ObjectMeta: metav1.ObjectMeta{Name: "pclass2"}, Value: int32(1000)}, metav1.CreateOptions{})
-		framework.ExpectEqual(err == nil || apierrors.IsAlreadyExists(err), true)
+		if err == nil && apierrors.IsAlreadyExists(err){
```
Comment (pohly) [inline test/e2e/apimachinery/resource_quota.go:1145]: See above.

File: test/e2e/apimachinery/resource_quota.go [hunk 6]
```diff
@@ -1183,7 +1189,9 @@ var _ = SIGDescribe("ResourceQuota [Feature:PodPriority]", func() {
 	ginkgo.It("should verify ResourceQuota's priority class scope (quota set to pod count: 1) against 2 pods with different priority class.", func() {
 
 		_, err := f.ClientSet.SchedulingV1().PriorityClasses().Create(context.TODO(), &schedulingv1.PriorityClass{ObjectMeta: metav1.ObjectMeta{Name: "pclass3"}, Value: int32(1000)}, metav1.CreateOptions{})
-		framework.ExpectEqual(err == nil || apierrors.IsAlreadyExists(err), true)
+		if err == nil && apierrors.IsAlreadyExists(err){
```
Comment (pohly) [inline test/e2e/apimachinery/resource_quota.go:1192]: See above.

File: test/e2e/apimachinery/resource_quota.go [hunk 7]
```diff
@@ -1229,10 +1237,14 @@ var _ = SIGDescribe("ResourceQuota [Feature:PodPriority]", func() {
 
 	ginkgo.It("should verify ResourceQuota's multiple priority class scope (quota set to pod count: 2) against 2 pods with same priority classes.", func() {
 		_, err := f.ClientSet.SchedulingV1().PriorityClasses().Create(context.TODO(), &schedulingv1.PriorityClass{ObjectMeta: metav1.ObjectMeta{Name: "pclass5"}, Value: int32(1000)}, metav1.CreateOptions{})
-		framework.ExpectEqual(err == nil || apierrors.IsAlreadyExists(err), true)
+		if err == nil && apierrors.IsAlreadyExists(err){
```
Comment (pohly) [inline test/e2e/apimachinery/resource_quota.go:1240]: See above.

File: test/e2e/apimachinery/resource_quota.go [hunk 8]
```diff
@@ -1229,10 +1237,14 @@ var _ = SIGDescribe("ResourceQuota [Feature:PodPriority]", func() {
...
+			framework.Failf("unexpected error while creating priority class: %v", err)
+		}
 
 		_, err = f.ClientSet.SchedulingV1().PriorityClasses().Create(context.TODO(), &schedulingv1.PriorityClass{ObjectMeta: metav1.ObjectMeta{Name: "pclass6"}, Value: int32(1000)}, metav1.CreateOptions{})
-		framework.ExpectEqual(err == nil || apierrors.IsAlreadyExists(err), true)
+		if err == nil && apierrors.IsAlreadyExists(err){
```
Comment (pohly) [inline test/e2e/apimachinery/resource_quota.go:1245]: See above.

File: test/e2e/apimachinery/resource_quota.go [hunk 9]
```diff
@@ -1284,7 +1296,9 @@ var _ = SIGDescribe("ResourceQuota [Feature:PodPriority]", func() {
 	ginkgo.It("should verify ResourceQuota's priority class scope (quota set to pod count: 1) against a pod with different priority class (ScopeSelectorOpNotIn).", func() {
 
 		_, err := f.ClientSet.SchedulingV1().PriorityClasses().Create(context.TODO(), &schedulingv1.PriorityClass{ObjectMeta: metav1.ObjectMeta{Name: "pclass7"}, Value: int32(1000)}, metav1.CreateOptions{})
-		framework.ExpectEqual(err == nil || apierrors.IsAlreadyExists(err), true)
+		if err == nil && apierrors.IsAlreadyExists(err){
```
Comment (pohly) [inline test/e2e/apimachinery/resource_quota.go:1299]: See above.

File: test/e2e/apimachinery/resource_quota.go [hunk 10]
```diff
@@ -1318,7 +1332,9 @@ var _ = SIGDescribe("ResourceQuota [Feature:PodPriority]", func() {
 	ginkgo.It("should verify ResourceQuota's priority class scope (quota set to pod count: 1) against a pod with different priority class (ScopeSelectorOpExists).", func() {
 
 		_, err := f.ClientSet.SchedulingV1().PriorityClasses().Create(context.TODO(), &schedulingv1.PriorityClass{ObjectMeta: metav1.ObjectMeta{Name: "pclass8"}, Value: int32(1000)}, metav1.CreateOptions{})
-		framework.ExpectEqual(err == nil || apierrors.IsAlreadyExists(err), true)
+		if err == nil && apierrors.IsAlreadyExists(err){
```
Comment (pohly) [inline test/e2e/apimachinery/resource_quota.go:1335]: See above.

File: test/e2e/apimachinery/resource_quota.go [hunk 11]
```diff
@@ -1357,7 +1373,9 @@ var _ = SIGDescribe("ResourceQuota [Feature:PodPriority]", func() {
 	ginkgo.It("should verify ResourceQuota's priority class scope (cpu, memory quota set) against a pod with same priority class.", func() {
 
 		_, err := f.ClientSet.SchedulingV1().PriorityClasses().Create(context.TODO(), &schedulingv1.PriorityClass{ObjectMeta: metav1.ObjectMeta{Name: "pclass9"}, Value: int32(1000)}, metav1.CreateOptions{})
-		framework.ExpectEqual(err == nil || apierrors.IsAlreadyExists(err), true)
+		if err == nil && apierrors.IsAlreadyExists(err){
```
Comment (pohly) [inline test/e2e/apimachinery/resource_quota.go:1376]: See above.

**Review by pohly — CHANGES_REQUESTED** [review]

**Review by pohly — APPROVED** [review]
/lgtm

**Comment by pohly** [comment]
/assign @caesarxuchao

For approval.

**Pull request merged**